src/core/aio.c

```diff
@@ -523,16 +527,61 @@ nni_aio_expire_add(nni_aio *aio)
 		// And, as we are the latest, kick the thing.
 		nni_cv_wake(&aio->a_expire_q->eq_cv);
 	}
+	*/
+
+	nni_aio_expire_q * eq = aio->a_expire_q;
+	if (eq->eq_len > eq->eq_cap * 2 / 3) {
```

Comment:
growing at 2/3?  why not just wait until you actually grow exactly to the point of needing it?

Reply:
also this division by three is both expensive and unnecessary.